File: DormFinder.Client/src/app/models/lead.model.ts
import { IContract } from 'src/app/models/contract.model';
import { IRoom } from './rooms.model';
import { IUser } from './user.model';

export interface IAddlead {
  id: number;
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  number: number;
  message: string;
  roomId: number;
  room: IRoom;
  createdAt: string;
  contract: IContract;
}

export interface Ilead {
  id: number;
  firstName: string;
  lastName: string;
  address: string;
  email: string;
  number: number;
  message: string;
  roomId: number;
  room: IRoom;
  createdAt: string;
  contract: IContract;
  userId: number;
}

File: DormFinder.Web/Models/LeadModel.cs
using System;
using DormFinder.Web.Rooms.Model;

namespace DormFinder.Web.Models
{
    public class LeadModel
    {
        public int Id { get; set; }

        public string Email { get; set; }

        public string Number { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Message { get; set; }

        public int RoomId { get; set; }

        public string Status { get; set; }

        public RoomDto Room { get; set; }

        public ContractModel Contract { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public int UserId { get; set; }

        public UserDto User { get; set; }
    }

    public class AddLeadModel
    {
        public string Email { get; set; }

        public string Number { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string Message { get; set; }

        public int RoomId { get; set; }
    }
}
